includes/Integrations/ShortPixel.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Class ShortPixel.
+ *
+ * @link      https://github.com/googleforcreators/web-stories-wp
+ *
+ * @copyright 2022 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ */
+
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\Integrations;
+
+use Google\Web_Stories\Service_Base;
+
+/**
+ * Class ShortPixel
+ */
+class ShortPixel extends Service_Base {
+
+	/**
+	 * Runs on instantiation.
+	 *
+	 * @since 1.23.0
+	 */
+	public function register(): void {
+		add_filter( 'shortpixel_image_urls', [ $this, 'image_urls' ], 10, 2 );
+	}
+
+	/**
+	 * Ensures page template urls bypass optimisation.
+	 *
+	 * @since 1.23.0
+	 *
+	 * @param string[] $urls  Urls that will be sent to optimisation.
+	 * @param int      $post_id Post ID.
+	 * @return string[] The filtered Urls.
+	 */
+	public function image_urls( $urls, $post_id ): array {
+		if ( str_contains( $urls[0], 'web-stories-page-template' ) ) {
```

Comment:
I'll adjust the code (just in case)

From what I see .... the urls contain are same at different sizes.

Also read that the post_id isn't always available so opted to check based on the URL string